test/e2e/e2e_test.go

```diff
@@ -142,26 +144,29 @@ func TestE2EFlow(t *testing.T) {
 		verifyResources(awaitility, currentMur, expectingMurConditions(toBeProvisioned()),
 			expectingUaConditions(toBeProvisioned(), coolStatus()))
 
-		extraMur = wait.NewHostAwaitility(awaitility).GetMasterUserRecord(extraMur.Name)
+		extraMur, err = wait.NewHostAwaitility(awaitility).WaitForMasterUserRecord(extraMur.Name)
+		require.NoError(t, err)
 		// TODO: verify expected condition when the member operator has a logic that updates NsTemplateSet and its status
 		verifyResources(awaitility, extraMur, nil, expectingUaConditions(toBeProvisioned()))
 	})
 
 	t.Run("delete MasterUserRecord and expect UserAccount to be deleted", func(t *testing.T) {
 		// given
-		currentMur := wait.NewHostAwaitility(awaitility).GetMasterUserRecord(mur.Name)
+		currentMur, err := wait.NewHostAwaitility(awaitility).WaitForMasterUserRecord(mur.Name)
+		require.NoError(t, err)
 
 		// when
-		err := awaitility.Client.Delete(context.TODO(), currentMur)
+		err = awaitility.Client.Delete(context.TODO(), currentMur)
 
 		// then
 		require.NoError(t, err)
 		t.Logf("MasterUserRecord '%s' deleted", mur.Name)
 
 		verifyDeletion(awaitility, currentMur)
-		assert.NoError(t, err)
+		require.NoError(t, err)
```

Comment:
ok, my apologies. It's my old habit of always using `require.NoError(...)` and I was too fast in making these changes. I'll revert them.

Reply:
right, it's a left over. But I wonder if for the sake of readability, we could pass `t` as the first arg, instead of carrying it in the `awaitility` object. WDYT?

Reply:
done in ae3a519